lib/versionmanager.js

```diff
@@ -11,6 +11,9 @@ const prompts = require('prompts');
 // keep order for setPrecision
 const DEFAULT_WILDCARD = '^';
 
+// track if all packages return error 404
+let allNotFoundErrors;
```

Comment:
This is still effectively global (to the file). To be clear, `let` (more broadly, mutability) *is a bad practice*. I recommend reading a little more about functional programming if this point is confusing for you.